Flaky test: k8sclusterreceiver TestCollectMetricData

See https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/5772813680/job/15648173206?pr=24925
https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/5772825398/job/15648195046?pr=24926
```
--- FAIL: TestCollectMetricData (0.00s)
    collector_test.go:121: 
        	Error Trace:	/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/receiver/k8sclusterreceiver/internal/collection/collector_test.go:121
        	Error:      	Received unexpected error:
        	            	the following errors occurred:
        	            	 -  resources are out of order: resource "map[k8s.node.name:test-node-2 k8s.node.uid:test-node-2-uid opencensus.resourcetype:k8s]" expected at index 12, found at index 13
        	            	 -  resources are out of order: resource "map[k8s.node.name:test-node-1 k8s.node.uid:test-node-1-uid opencensus.resourcetype:k8s]" expected at index 13, found at index 12
        	Test:       	TestCollectMetricData
```

--- a/receiver/k8sclusterreceiver/internal/collection/collector_test.go
+++ b/receiver/k8sclusterreceiver/internal/collection/collector_test.go
@@ -118,5 +118,5 @@ func TestCollectMetricData(t *testing.T) {
 	m2 := dc.CollectMetricData(time.Now())
 
 	// Second scrape should be the same as the first one except for the timestamp.
-	assert.NoError(t, pmetrictest.CompareMetrics(m1, m2, pmetrictest.IgnoreTimestamp()))
+	assert.NoError(t, pmetrictest.CompareMetrics(m1, m2, pmetrictest.IgnoreTimestamp(), pmetrictest.IgnoreResourceMetricsOrder()))
 }

